### PR TITLE
feat: debt switch

### PR DESCRIPTION
**Pull request by defispartan**
## General Changes

- Incorporates the debt switch adapter contracts, see more details here: https://governance.aave.com/t/bgd-aave-debt-swap-adapter-contract/12738
- Adds in debt switch support to the following markets:
  - Eth v2
  - Polygon v2 and v3
  - Avalanche v2 and v3
  - Optimism v3
  - Arbitrum v3

---

## Author Checklist

Please ensure you, the author, have gone through this checklist to ensure there is an efficient workflow for the reviewers.

- [ ]  The base branch is set to `main`
- [ ]  The title is using [Conventional Commit](https://www.conventionalcommits.org/en/v1.0.0/) formatting
- [ ]  The Github issue has been linked to the PR in the Development section
- [ ]  The General Changes section has been filled out
- [ ]  Developer Notes have been added (optional)

**If the PR is ready for review:**

- [ ]  The PR is in `Open` state and not in `Draft` mode
- [ ]  The `Ready for Dev Review` label has been added

## Reviewer Checklist

Please ensure you, as the reviewer(s), have gone through this checklist to ensure that the code changes are ready to ship safely and to help mitigate any downstream issues that may occur.

- [ ]  End-to-end tests are passing without any errors
- [ ]  Code style generally follows existing patterns
- [ ]  Code changes do not significantly increase the application bundle size
- [ ]  If there are new 3rd-party packages, they do not introduce potential security threats
- [ ]  If there are new environment variables being added, they have been added to the `.env.example` file as well as the pertinant `.github/actions/*` files
- [ ]  There are no CI changes, or they have been approved by the DevOps and Engineering team(s)
- [ ]  Code changes have been quality checked in the ephemeral URL
- [ ]  QA verification has been completed
- [ ]  There are two or more approvals from the core team
- [ ]  Squash and merge has been checked

### CODE REVIEW

**Comment by defispartan** [comment]
Link T-9392

File: src/store/poolSlice.ts [hunk 1]
```diff
@@ -61,6 +63,22 @@ export type PoolReserve = {
   userReserves?: UserReserveDataHumanized[];
 };
 
+//TODO: temp, move to DebtSwitchActions once types are finalized
+interface DebtSwitchActionsProps {
```
Comment (grothem) [inline src/store/poolSlice.ts:67]: Looks like there's a lot of overlap between this interface and the one that's already in `DebtSwitchActions`. Probably just need to consolidate the two in that component.

Comment (defispartan) [reply src/store/poolSlice.ts]: Consolidated to DebtSwitchActionProps

**Comment by MareskoY** [comment]
mixpannel events also tested, looks good

**Review by grothem — APPROVED** [review]

**Review by foodaka — APPROVED** [review]

**Pull request merged**